src/platform/datapath_kqueue.c

```diff
@@ -721,126 +724,208 @@ CxPlatSocketContextInitialize(
     }
 
     //
-    // Set non blocking mode
+    // Set dual (IPv4 & IPv6) socket mode unless we operate in pure IPv4 mode
     //
-    Flags =
-        fcntl(
+    if (!ForceIpv4) {
+        Option = FALSE;
+        Result =
+            setsockopt(
+                SocketContext->SocketFd,
+                IPPROTO_IPV6,
+                IPV6_V6ONLY,
+                (const void*)&Option,
+                sizeof(Option));
+        if (Result == SOCKET_ERROR) {
+            Status = errno;
+            QuicTraceEvent(
+                DatapathErrorStatus,
+                "[data][%p] ERROR, %u, %s.",
+                Binding,
+                Status,
+                "setsockopt(IPV6_V6ONLY) failed");
+            goto Exit;
+        }
+    }
+
+    //
+    // The port may be shared across processors.
+    // Even if not, this is probably cheap.
```

Comment:
I don't remember if this was already here in the code, but if we aren't sharing it with ourselves (i.e. in the same process) then we shouldn't share it with others by default.

Reply:
this is leftover from existing code (just moved). I don't think OS will distinguish between sharing with same process vs sharing with others. This simply allows second socket to bind with the same parameters. This may not be needed here as it seems like the purpose on Linux is load balancing on multiple cores.

Reply:
I know on TCP this is used a lot more often so the port can be reused without waiting for the socket to timeout, even in normal processes. For UDP however, I don't think its an issue, and we can likely remove this. Let me check real quick.

Reply:
Ok. So for UDP its for multicast support, so we don't need it. So I don't think its needed here, or for client sockets on the linux side, but thats a separate issue.

Reply:
this was originally in the code. removed since all the tests are passing without it.